packages/legend-graph/src/models/DomainModifierHelper.ts

```diff
@@ -0,0 +1,373 @@
+    _m.upperBound = val;
+  },
+);
+
+// Constraint
+export const setConstraintName = action(
+  (_constraint: Constraint, name: string): void => {
+    _constraint.name = name;
+  },
+);
+export const setConstraintFunctionDefinition = action(
+  (_constraint: Constraint, lambda: RawLambda): void => {
+    _constraint.functionDefinition = lambda;
+  },
+);
+
+// Profile
+export const addProfileTag = action((profile: Profile, value: Tag): void => {
+  addUniqueEntry(profile.tags, value);
+});
+export const deleteProfileTag = action((profile: Profile, value: Tag): void => {
+  deleteEntry(profile.tags, value);
+});
+export const addProfileStereotype = action(
+  (profile: Profile, value: Stereotype): void => {
+    addUniqueEntry(profile.stereotypes, value);
+  },
+);
+export const deleteProfileStereotype = action(
+  (profile: Profile, value: Stereotype): void => {
+    deleteEntry(profile.stereotypes, value);
+  },
+);
+
+// Function
+export const deleteFunctionParameter = action(
+  (_func: ConcreteFunctionDefinition, val: RawVariableExpression): void => {
+    deleteEntry(_func.parameters, val);
+  },
+);
+export const addFunctionParameter = action(
+  (_func: ConcreteFunctionDefinition, val: RawVariableExpression): void => {
+    addUniqueEntry(_func.parameters, val);
+  },
+);
+export const setFunctionReturnType = action(
+  (_func: ConcreteFunctionDefinition, val: Type): void => {
+    setPackageableElementReferenceValue(_func.returnType, val);
+  },
+);
+export const setFunctionReturnMultiplicity = action(
+  (_func: ConcreteFunctionDefinition, val: Multiplicity): void => {
+    _func.returnMultiplicity = val;
+  },
+);
+
+// Enumeration
+export const setEnumName = action((val: Enum, value: string): void => {
+  val.name = value;
+});
+export const addEnumValue = action(
+  (enumeration: Enumeration, value: Enum): void => {
+    addUniqueEntry(enumeration.values, value);
+  },
+);
+export const deleteEnumValue = action(
+  (enumeration: Enumeration, value: Enum): void => {
+    deleteEntry(enumeration.values, value);
+  },
+);
+export const setEnumValueReferenceValue = action(
+  (ref: EnumValueReference, value: Enum): void => {
+    ref.value = value;
+    setPackageableElementReferenceValue(ref.ownerReference, value.owner);
+  },
+);
+
+// Measure
+export const setMeasureCanonicalUnit = action(
+  (_measure: Measure, unit: Unit): void => {
+    _measure.canonicalUnit = unit;
+  },
+);
+export const setUnitConversionFunction = action(
+  (unit: Unit, lambda: RawLambda): void => {
+    unit.conversionFunction = lambda;
+  },
+);
+
+// Package
+
+export const addPackageChild = action(
+  (parent: Package, value: PackageableElement): void => {
+    // NOTE: here we directly push the element to the children array without any checks rather than use `addUniqueEntry` to improve performance.
+    // Duplication checks should be handled separately
+    parent.children.push(value);
+  },
+);
+
+export const addPackageElement = action(
```

Comment:
After reading a while, I guess the rule of thumb for me is I don't want to touch modifiers with a bit of logic, e.g. this method or `setValue` for references.